### PR TITLE
Marvel Client API assignment 

### PR DESCRIPTION
**Pull request by aguazul-marco**

### CODE REVIEW

**Review by jboursiquot — CHANGES_REQUESTED** [review]
You've submitted a duplicate of the example I did in class.
Read the assignment again: https://fifth-bottom-224.notion.site/Build-a-Marvel-Client-669c3b397f844a96a6e6d4659ca20cb3

And try to work through the solution from scratch, only using my example as a reference so that it does not look like you simply copied my code.

Let me know if you need some one-on-one assistance.

**Comment by aguazul-marco** [comment]
Code Output:
```
([]marvel.Character) (len=5 cap=6) {
 (marvel.Character) {
  ID: (int) 1011334,
  Name: (string) (len=7) "3-D Man",
  Description: (string) ""
 },
 (marvel.Character) {
  ID: (int) 1017100,
  Name: (string) (len=12) "A-Bomb (HAS)",
  Description: (string) (len=302) "Rick Jones has been Hulk's best bud since day one, but now he's more than a friend...he's a teammate! Transformed by a Gamma energy explosion, A-Bomb's thick, armored skin is just as strong and powerful as it is blue. And when he curls into action, he uses it like a giant bowling ball of destruction! "
 },
 (marvel.Character) {
  ID: (int) 1009144,
  Name: (string) (len=6) "A.I.M.",
  Description: (string) (len=61) "AIM is a terrorist organization bent on destroying the world."
 },
 (marvel.Character) {
  ID: (int) 1010699,
  Name: (string) (len=11) "Aaron Stack",
  Description: (string) ""
 },
 (marvel.Character) {
  ID: (int) 1009146,
  Name: (string) (len=26) "Abomination (Emil Blonsky)",
  Description: (string) (len=236) "Formerly known as Emil Blonsky, a spy of Soviet Yugoslavian origin working for the KGB, the Abomination gained his powers after receiving a dose of gamma radiation similar to that which transformed Bruce Banner into the incredible Hulk."
 }
}
```

File: cmd/marvel-client/types.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package main
```
Comment (jboursiquot) [inline cmd/marvel-client/types.go:1]: This should be part of package `marvel`, not `main`.

File: cmd/marvel-client/main.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	privKey := os.Getenv("MARVEL_PRIVATE_KEY")
+
+	client := marvel.NewClient(pubKey, privKey)
+
+	character, err := client.GetCharacter(5)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	spew.Dump(character)
```
Comment (jboursiquot) [inline cmd/marvel-client/main.go:28]: The assignment calls for listing out the name and description of the characters, not a dump.

File: marvel/marvel.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	return hex.EncodeToString(hash[:])
+}
+
+func (c *Client) signURL(url string) string {
+	ts := time.Now().Unix()
+	hash := c.md5Hash(ts)
+	return fmt.Sprintf("%s&ts=%d&apikey=%s&hash=%s", url, ts, c.PubKey, hash)
+}
+
+func (c *Client) GetCharacter(limit int) ([]Character, error) {
```
Comment (jboursiquot) [inline marvel/marvel.go:65]: You're retrieving a list of characters -- plural. This method name could use a tweak, don't you agree?

**Review by jboursiquot — CHANGES_REQUESTED** [review]

**Comment by aguazul-marco** [comment]
Output:
```
Character: 3-D Man
Description: 

Character: A-Bomb (HAS)
Description: Rick Jones has been Hulk's best bud since day one, but now he's more than a friend...he's a teammate! Transformed by a Gamma energy explosion, A-Bomb's thick, armored skin is just as strong and powerful as it is blue. And when he curls into action, he uses it like a giant bowling ball of destruction! 

Character: A.I.M.
Description: AIM is a terrorist organization bent on destroying the world.

Character: Aaron Stack
Description: 

Character: Abomination (Emil Blonsky)
Description: Formerly known as Emil Blonsky, a spy of Soviet Yugoslavian origin working for the KGB, the Abomination gained his powers after receiving a dose of gamma radiation similar to that which transformed Bruce Banner into the incredible Hulk.
```

**Review by jboursiquot — APPROVED** [review]
Well done.

**Pull request merged**